### PR TITLE
Move omni_getcrowdsale amountraised field into verbose output and sort participants

### PR DESCRIPTION
**Pull request by zathras-crypto**
Note: we could leave amount raised in the non-verbose output, but that would mean calculating all participants when non-verbose which is a slow and expensive call.

### CODE REVIEW

File: src/omnicore/rpc.cpp [hunk 1]
```diff
@@ -696,28 +694,24 @@ Value omni_getcrowdsale(const Array& params, bool fHelp)
...
-            int64_t amountSent = it->second.at(0);
-
-            amountRaised += amountSent;
+            amountRaised += it->second.at(0);
             participanttx.push_back(Pair("txid", txid));
-            participanttx.push_back(Pair("amountsent", FormatMP(sp.property_desired, amountSent)));
-            participanttx.push_back(Pair("participanttokens", FormatMP(propertyId, userTokens)));
-            participanttx.push_back(Pair("issuertokens", FormatMP(propertyId, issuerTokens)));
-            participanttxs.push_back(participanttx);
+            participanttx.push_back(Pair("amountsent", FormatMP(sp.property_desired, (int64_t)it->second.at(0))));
```
Comment (dexX7) [inline src/omnicore/rpc.cpp]: No need to cast `(int64_t)` for `it->second`, it's already `int64_t`.

File: src/omnicore/rpc.cpp [hunk 2]
```diff
@@ -696,28 +694,24 @@ Value omni_getcrowdsale(const Array& params, bool fHelp)
...
+            participanttx.push_back(Pair("participanttokens", FormatMP(propertyId, (int64_t)it->second.at(2))));
+            participanttx.push_back(Pair("issuertokens", FormatMP(propertyId, (int64_t)it->second.at(3))));
+            sortMap.insert(make_pair((int64_t)it->second.at(1), participanttx));
```
Comment (dexX7) [inline src/omnicore/rpc.cpp]: Nit: `std::make_pair`.. :)

**Comment by dexX7** [comment]
> Note: we could leave amount raised in the non-verbose output, but that would mean calculating all participants when non-verbose which is a slow and expensive call.

It's probably okay to move it into the non-verbose output, because the whole SP, including database, is already deserialized at that point, and all values in memory.

A likely more signifiant improvement could be to replace the `FormatMP()` calls with `FormatByType(int64_t amount, uint16_t propertyType)` (where `propertyType` is stored once), because each time `FormatMP` is used, the SP is fetched over and over again from the DB to determine divisibility.

**Comment by zathras-crypto** [comment]
Sorry dude, I did mean to include your feedback.  You're quite correct, it's really poor performance to load the SP on every iteration of the recipients loop - once that's modified the performance is very quick so I agree the calculation of ```amountraised``` can be moved out of the verbose output.

The sorting methodology is also floored, since I used a map ordered by blockheight, block is also the key resulting in only one crowdsale purchase being stored per block - you can see this from checking the amountraised against the total amountsent.

```
$ src/omnicore-cli getcrowdsale_MP 3 | grep amountraised
    "amountraised" : "93249.21601551",

$ src/omnicore-cli getcrowdsale_MP 3 true | grep amountsent
            "amountsent" : "100.00000000",
            "amountsent" : "8.07394800",
            "amountsent" : "2.04500000",
            "amountsent" : "1000.00000000",
            "amountsent" : "110.00000000",
            "amountsent" : "189.55000000",
            "amountsent" : "5.00000000",
            "amountsent" : "50.00000000",
            "amountsent" : "1.99882495",
            "amountsent" : "7.00000000",
            "amountsent" : "175.07666050",
            "amountsent" : "0.20000000",
            "amountsent" : "305.51100000",
            "amountsent" : "100.00000000",
            "amountsent" : "6.19185495",
            "amountsent" : "7.00000000",
            "amountsent" : "50.00000000",
            "amountsent" : "70.44000000",
            "amountsent" : "45.00000000",
            "amountsent" : "0.80000000",
            "amountsent" : "100.00000000",
            "amountsent" : "20.00000000",
            "amountsent" : "20.00000000",
            "amountsent" : "40.00000000",
```

I've made the changes fixes here 8f5241699bce83b617226acb05bdfda7f2d0a748 :)

**Comment by dexX7** [comment]
> Sorry dude, I did mean to include your feedback.

Oh my bad, I'm really sorry for the rushed merge then!

> The sorting methodology is also flawed ...

I see it now.

**Pull request merged**